### ISSUE TITLE
DR-Update GET endpoint on Frontend to consume API Gateway endpoint

### ISSUE THREAD

**Issue by charlottekies** [RubySpeeders/Drum-Roulette#78]
### User Story

As a developer, I want to be able to invoke the API Gateway Endpoints, so that we can retire the Spring Boot resource server
___
### General Description
 
The scope of this ticket involves changing the GET endpoint in the front end files from the endpoint currently used, to the endpoint defined in API Gateway. 
___
### Technical Steps
 

- [ ] Change the endpoint to the endpoint defined in API Gateway 
- [ ] Data returned from the API Gateway may have a different JSON structure than that from the Spring Boot server. Update model classes/structs to match most-recent JSON structure defined in WIKI API Documentation 
___
### Acceptance Criteria
- [x] Front end successfully gets musician data from the backend 

___

**Comment by RubySpeeders** [comment]
2

**Comment by charlottekies** [comment]
@RubySpeeders @cheesehero112 @anquail 

TLDR; 
Replace /musicians with /musicians/navy/concert  for the NAVY button 

As we are updating the API, the original GET endpoint with the path ending in /musicians now retrieves ALL musicians in the database. I suggest that whoever takes on this ticket ensures that when the user clicks on the NAVY button that only the NAVY band musicians are returned. (/musicians/navy/concert) 

Additionally, we may want to add a page to allow the user to select whether they want to include ALL navy band members, or just the navy concert band members. I understand sometimes people from the commodores fill in for certain ceremonies. It might be nice to have the option to include members from other ensembles within the same branch.

**Comment by RubySpeeders** [comment]
@charlottekies 

I'm wondering if we could change the order of the words to /navy/musicians/concert? And for the other branches, /{militarybranch}/musicians/{bandname}. If that's not an easy fix, I completely understand. My thinking is that it makes more sense to separate into the military branches first, since that is what happens on the landing page.

As for the difference between all navy and concert navy musicians, I think we should put that in the backlog as a stretch goal! I think that would be something to keep in mind, definitely, but will require lots of design and collab on how to make that clear to the user.

**Comment by charlottekies** [comment]
> @charlottekies
> 
> I'm wondering if we could change the order of the words to /navy/musicians/concert? And for the other branches, /{militarybranch}/musicians/{bandname}. If that's not an easy fix, I completely understand. My thinking is that it makes more sense to separate into the military branches first, since that is what happens on the landing page.
> 
> As for the difference between all navy and concert navy musicians, I think we should put that in the backlog as a stretch goal! I think that would be something to keep in mind, definitely, but will require lots of design and collab on how to make that clear to the user.

@RubySpeeders I thought about this and @anquail please weigh in if you have an opinion-- 

I think we shouldn't switch to match the navigation of the website. Here's why:
I think that the endpoint in a RESTful API is typically defined by the _thing_ you are retrieving. I think it should start with musicians because what we are retrieving is a group of musicians. Then the subsequent parts of the endpoint further specify the selection criteria. I think it would be a more traditional pattern to start with musicians, because we're GETting musicians. 

Would it be ok in this case to not make this change if this reasoning makes sense?
